bin/tx-prover/Cargo.toml

```diff
@@ -11,30 +11,41 @@ repository.workspace = true
 rust-version.workspace = true
 edition.workspace = true
 
+[[bin]]
+name = "prover-service"
+path = "src/main.rs"
+required-features = ["std"]
+
 [features]
-testing = ["miden-objects/testing", "miden-lib/testing", "miden-tx/testing"]
+std = ["miden-objects/std", "miden-tx/std", "dep:tokio", "dep:tonic-web", "dep:tokio-stream", "dep:axum",  "dep:tracing", "dep:tracing-subscriber"]
+async = ["miden-tx/async"]
+wasm = ["dep:tonic-web-wasm-client", "dep:getrandom", "getrandom/js"]
```

Comment:
Question: would it be a bad idea to determine if we are building for WASM based on the build target (rather than having an explicit feature for it)? How do we do it in the client?

Reply:
I thought about this, but then realized that maybe it's better to generalize to WASM targets instead of just supporting `wasm32-unknown-unknown`. On a deeper look though, it seems that the [tonic WASM client](https://github.com/devashishdxt/tonic-web-wasm-client?tab=readme-ov-file#building) is built just for this target, so I'll change it to be based on target architecture.

On the client you enable structs based on features (`idxdb` for the web store implementation, for example, or `web-tonic` for the analogous RPC struct)